src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -50,6 +53,115 @@ private protected override int GetSpanStart(SyntaxNode declaration)
+                if (symbol is not null && symbol.IsAwaitableNonDynamic(semanticModel, position))
+                {
+                    var parentOfAwaitable = potentialAwaitableExpression.Parent;
+                    if (parentOfAwaitable is not AwaitExpressionSyntax)
+                    {
+                        // We have a awaitable type left of the dot, that is not yet awaited.
+                        // We need to check if await is valid at the insertion position.
+                        var syntaxContextAtInsertationPosition = syntaxContext.GetLanguageService<ISyntaxContextService>().CreateContext(
+                            syntaxContext.Workspace, syntaxContext.SemanticModel, potentialAwaitableExpression.SpanStart, cancellationToken);
+                        if (syntaxContextAtInsertationPosition.IsAwaitKeywordContext())
+                        {
+                            return IsConfigureAwaitable(symbol)
+                                ? DotAwaitContext.AwaitAndConfigureAwait
+                                : DotAwaitContext.AwaitOnly;
+                        }
+                    }
+                }
+            }
+
+            return DotAwaitContext.None;
+
+            static ITypeSymbol? GetTypeSymbolOfExpression(SemanticModel semanticModel, SyntaxNode potentialAwaitableExpression, CancellationToken cancellationToken)
+            {
+                if (potentialAwaitableExpression is MemberAccessExpressionSyntax memberAccess)
+                {
+                    var memberAccessExpression = memberAccess.Expression.WalkDownParentheses();
+                    var symbol = semanticModel.GetSymbolInfo(memberAccessExpression, cancellationToken).Symbol;
+                    if (symbol is INamedTypeSymbol) // e.g. Task.$$
+                    {
+                        return null;
+                    }
+
+                    return
+                        symbol?.GetSymbolType() ??
+                        symbol?.GetMemberType() ??
+                        // Some expressions don't have a symbol (e.g. (o as Task).$$), but GetTypeInfo finds the right type.
+                        semanticModel.GetTypeInfo(memberAccessExpression, cancellationToken).Type;
+                }
+
+                if (potentialAwaitableExpression is ExpressionSyntax expression)
+                {
+                    if (expression.ShouldBeTreatedAsTypeInsteadOfExpression(semanticModel, out _, out var container))
+                        return container;
+                }
+
+                return semanticModel.GetTypeInfo(potentialAwaitableExpression, cancellationToken).Type;
+            }
+
+            static bool IsConfigureAwaitable(ITypeSymbol symbol)
+            {
+                const string TaskNameSpace = "System.Threading.Tasks";
+                return symbol.OriginalDefinition.ToSignatureDisplayString() is $"{TaskNameSpace}.Task" or $"{TaskNameSpace}.Task<TResult>";
```

Comment:
That's probably wrong, but does the trick for the moment. Unfortunately, `WellKnownTypes` from the compiler layer are not accessible here.